src/backend/aspen/api/conftest.py

```diff
@@ -0,0 +1,122 @@
+import random
+import string
+from dataclasses import dataclass
+from functools import partial
+from typing import AsyncGenerator
+
+import pytest
+from fastapi import Depends, FastAPI, Request
+from httpx import AsyncClient
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.api.auth import get_auth_user, setup_userinfo
+from aspen.api.deps import get_db
+from aspen.api.main import get_app
+from aspen.database import connection as aspen_connection
+from aspen.database import schema
+from aspen.database.connection import init_async_db
+
+USERNAME = "user_rw"
+PASSWORD = "password_rw"
+INITIAL_DATABASE = "postgres"
+
+pytestmark = pytest.mark.anyio
+
```

Comment:
Most of the code in this file is ~identical to the code in `aspen/test_infra` except that it's been updated to work with an async db interface